adapters/amqp-vertx-base/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -398,13 +445,48 @@ protected void onConnectRequest(final ProtonConnection con) {
         });
     }
 
+    /**
+     * Actively closes the connection to the device.
+     */
+    private void closeDeviceConnection(final ProtonConnection con, final String reason, final boolean sendDisconnectedEvent) {
+        final Device authenticatedDevice = getAuthenticatedDevice(con);
+        final Span span = newSpan("close device connection", authenticatedDevice, getTraceSamplingPriority(con));
```

Comment:
can we finish the Span in this method as well? FMPOV it makes the code much easier to read and prevents loose ends ...